File: wordDictionary.py
from collections import defaultdict
class WordDictionary:
    def __init__(self):
        self.words = defaultdict(set)
    def addWord(self, word):
        self.words[len(word)].add(word)
        
    def search(self, word):
        for i in self.words[len(word)]:
            for x,c in enumerate(word):
                print(i[x])
                if c not in ['.',  i[x]]:
                    break
            # any_mismatch = any(word[x] != '.' and word[x] != i[x]  for x in range(len(word)))
                else:
                    return True
        return False
		       
    
obj=WordDictionary()
obj.addWord("bad")
obj.addWord("dad")
obj.addWord("mad")
obj.addWord("hello")
print(obj.words)
print(obj.search("pad"))
print(obj.search("bad"))
print(obj.search(".ad"))
print(obj.search("b.."))

File: wordDictionary.js
class WordDictionary{
    constructor(){
        this.words={}
    }
    addWord(word){
        
        if (word.length in this.words){
            this.words[word.length].add(word)
        }
        else{
            this.words[word.length]=new Set()
            this.words[word.length].add(word)
        }
    }
    search(word){
        
        for (let i=0;i<Array.from(this.words[word.length]).length; i++){
            for (let j in word){
               
                if (!['.',  i[j]].includes(word[j])){
                    
                    break
                }
                else{
                    return true
                }
             }
        }
        return false
    }
}

obj=new WordDictionary()
console.log(obj.words)
obj.addWord("bad")
obj.addWord("dad")
obj.addWord("mad")
obj.addWord("hello")
console.log(obj.words)
console.log(obj.search("pad"))
console.log(obj.search("bad"))
console.log(obj.search(".ad"))
console.log(obj.search("b.."))